src/test/java/com/auth0/utils/tokens/SignatureVerifierTest.java

```diff
@@ -0,0 +1,213 @@
+        exception.expect(IdTokenValidationException.class);
+        exception.expectMessage("Error retrieving public key");
+        exception.expectCause(allOf(
+                instanceOf(PublicKeyProviderException.class),
+                hasProperty("cause", instanceOf(IOException.class))
+        ));
+
+        SignatureVerifier verifier = SignatureVerifier.forRS256(new PublicKeyProvider() {
+            @Override
+            public RSAPublicKey getPublicKeyById(String keyId) throws PublicKeyProviderException {
+                throw new PublicKeyProviderException("error", new IOException("error reading file"));
+            }
+        });
+        verifier.verifySignature(RS_JWT);
+    }
+
+    @Test
+    public void failsWithNullVerifier() {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage("'verifier' cannot be null");
+        new NullVerifier();
+    }
+
+    private PublicKeyProvider getRSProvider(String rsaPath) throws Exception {
+        return new PublicKeyProvider() {
+            @Override
+            public RSAPublicKey getPublicKeyById(String keyId) throws PublicKeyProviderException {
+                try {
+                    return readPublicKeyFromFile(rsaPath);
+                } catch (IOException ioe) {
+                    throw new PublicKeyProviderException("Error reading public key", ioe);
+                }
+            };
+        };
+    }
+
+    private static RSAPublicKey readPublicKeyFromFile(final String path) throws IOException {
+        Scanner scanner = null;
+        PemReader pemReader = null;
+        try {
+            scanner = new Scanner(Paths.get(path));
+            if (scanner.hasNextLine() && scanner.nextLine().startsWith("-----BEGIN CERTIFICATE-----")) {
+                FileInputStream fs = new FileInputStream(path);
+                CertificateFactory fact = CertificateFactory.getInstance("X.509");
+                X509Certificate cer = (X509Certificate) fact.generateCertificate(fs);
+                PublicKey key = cer.getPublicKey();
+                fs.close();
+                return (RSAPublicKey) key;
+            } else {
+                pemReader = new PemReader(new FileReader(path));
+                byte[] keyBytes = pemReader.readPemObject().getContent();
+                KeyFactory kf = KeyFactory.getInstance("RSA");
+                EncodedKeySpec keySpec = new X509EncodedKeySpec(keyBytes);
+                return (RSAPublicKey) kf.generatePublic(keySpec);
+            }
+        } catch (Exception e) {
+            throw new IOException("Couldn't parse the RSA Public Key / Certificate file.", e);
+        } finally {
+            if (scanner != null) {
+                scanner.close();
+            }
+            if (pemReader != null) {
+                pemReader.close();
+            }
+        }
+    }
+
+    private static class NullVerifier extends SignatureVerifier {
```

Comment:
will probably go away

Reply:
Just added for the test to verify the null check of the `SignatureVerifier`.